squad-server/plugins/chat-commands.js

```diff
@@ -36,19 +36,31 @@ export default class ChatCommands extends BasePlugin {
     };
   }
 
-  constructor(server, options, optionsRaw) {
-    super(server, options, optionsRaw);
+  async init () {
+    this.options.commands.forEach(commandSettings => {
+      if (commandSettings.type === 'broadcast')
+        this.server.on(`CHAT_COMMAND:${commandSettings.command}`, this.handleChatCommandBroadcast.bind(this, commandSettings));
```

Comment:
What for ? You only use .bind(this) to pass parameters for the scope being used in the handle since it looses its class scope.
There being lazy i used this all over, but normaly you would use only necessary info.

Reply:
`this.x.bind(this)` will not equal `this.x` to my understanding as it returns a new function. So here when you add and remove the event listener with and without `.bind(this)` respectively it will not work as they reference different functions. It's pretty common to therefore assign the bind in the constructor, as seen in React apps.

Reply:
This seem to be the case.
Still i would rather see every function being bound to respective input ex `x.bind(commandSettings)` instead and every listener saved in some kind of pool.